### PR TITLE
Workflow revert

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements #583 - Revert workflow

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowHistoryController.scala [hunk 1]
```diff
@@ -33,6 +33,11 @@ class WorkflowHistoryController @Inject()(workflowHistoryService: WorkflowHistor
     workflowHistoryService.getHistoryForWorkflow(workflowId).toJava.toCompletableFuture
   }
 
+  @GetMapping(path = Array("/historyWorkflow"))
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowHistoryController.scala]: I think it should be `/workflowFromHistory`, similar to the endpoint below. `/historyWorkflow` is confusing, especially because there is also `/workflowHistory`. Another possibility is `/workflowHistoryEntry`

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowHistoryService.scala [hunk 2]
```diff
@@ -25,6 +25,7 @@ trait WorkflowHistoryService {
   val workflowHistoryRepository: WorkflowHistoryRepository
 
   def getHistoryForWorkflow(workflowId: Long)(implicit ec: ExecutionContext): Future[Seq[History]]
+  def getHistoryWorkflow(workflowHistoryId: Long)(implicit ec: ExecutionContext): Future[WorkflowJoined]
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowHistoryService.scala]: Same here. I suggest `getWorkflowFromHistory`, or maybe something like `getWorkflowHistoryEntry`

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/WorkflowHistoryRepository.scala [hunk 3]
```diff
@@ -31,6 +31,7 @@ trait WorkflowHistoryRepository extends Repository {
   private[persistance] def delete(workflow: WorkflowJoined, user: String)(implicit ec: ExecutionContext): DBIO[Long]
 
   def getHistoryForWorkflow(workflowId: Long)(implicit ec: ExecutionContext): Future[Seq[History]]
+  def getHistoryWorkflow(workflowHistoryId: Long)(implicit ec: ExecutionContext): Future[WorkflowJoined]
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/persistance/WorkflowHistoryRepository.scala]: Same here

File: ui/src/app/services/workflowHistory/workflow-history.service.ts [hunk 4]
```diff
@@ -34,6 +35,14 @@ export class WorkflowHistoryService {
       .pipe(map((_) => _.body));
   }
 
+  getHistoryWorkflow(id: number): Observable<WorkflowJoinedModel> {
```
Comment (kevinwallimann) [inline ui/src/app/services/workflowHistory/workflow-history.service.ts]: Also here the naming

File: ui/src/app/stores/workflows/workflows.reducers.ts [hunk 5]
```diff
@@ -496,6 +496,35 @@ export function workflowsReducer(state: State = initialState, action: WorkflowsA
...
+      };
+    case WorkflowsActions.REVERT_WORKFLOW_SUCCESS:
+      return {
+        ...state,
+        workflowAction: {
+          ...state.workflowAction,
+          loading: false,
+          workflow: undefined,
+          workflowForForm: action.payload,
+          backendValidationErrors: [],
```
Comment (kevinwallimann) [inline ui/src/app/stores/workflows/workflows.reducers.ts:517]: Why is this needed?

File: ui/src/app/stores/workflows/workflows.reducers.ts [hunk 6]
```diff
@@ -496,6 +496,35 @@ export function workflowsReducer(state: State = initialState, action: WorkflowsA
...
+          loading: true,
+        },
+      };
+    case WorkflowsActions.REVERT_WORKFLOW_SUCCESS:
+      return {
+        ...state,
+        workflowAction: {
+          ...state.workflowAction,
+          loading: false,
+          workflow: undefined,
```
Comment (kevinwallimann) [inline ui/src/app/stores/workflows/workflows.reducers.ts:515]: Does this not have to be `action.payload`?

File: ui/src/app/stores/workflows/workflows.reducers.ts (hunk 5 above)
Comment (jozefbakus) [reply ui/src/app/stores/workflows/workflows.reducers.ts:517]: We show some errors this way

File: ui/src/app/stores/workflows/workflows.reducers.ts (hunk 6 above)
Comment (jozefbakus) [reply ui/src/app/stores/workflows/workflows.reducers.ts:515]: Not really, You would not be able to update a workflow via form because we compare workflow and workflowForForm

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowHistoryController.scala (hunk 1 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowHistoryController.scala]: Fixed

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowHistoryService.scala (hunk 2 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowHistoryService.scala]: Fixed

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/WorkflowHistoryRepository.scala (hunk 3 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/persistance/WorkflowHistoryRepository.scala]: Fixed

File: ui/src/app/services/workflowHistory/workflow-history.service.ts (hunk 4 above)
Comment (jozefbakus) [reply ui/src/app/services/workflowHistory/workflow-history.service.ts]: Fixed

**Review by kevinwallimann — APPROVED** [review]
